lib/practice_exercises.rb

```diff
@@ -1,13 +1,43 @@
+require "pry"
+# Given nums = [0,0,1,1,1,2,2,3,3,4]
+# Your function should return length = 5, 
+# with the first five elements of nums being modified to 0, 1, 2, 3, and 4 respectively.
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n) - depends of the length of the list
+# Space Complexity: O(1) - doens't create another array
 def remove_duplicates(list)
-  raise NotImplementedError, "Not implemented yet"
-end
+  i = 0
+  while i < list.length
+    if list[i] == list[i+1]
+      list.delete_at(i)
```

Comment:
`delete_at` shifts all subsequent  elements left one index each.  This is an O(n) method.  Since it's nested inside a loop running n times, this means the method is an O(n<sup>2</sup>) algorithm.